src/mewbot/io/desktop_notification.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+from typing import Optional, Set, Sequence, Type, Any
+
+import dataclasses
+import logging
+import sys
+import subprocess
+
+from mewbot.core import OutputEvent
+from mewbot.api.v1 import IOConfig, Input, Output
+
+# Input for this class is theoretically possible and would be desirable - would allow mewbot to
+# trigger on an arbitrary desktop notifications.
+# Development ongoing
+
+
+@dataclasses.dataclass
+class DesktopNotificationOutputEvent(OutputEvent):
+    """
+    In most notification systems, you need a title and a body.
+    """
+
+    title: str
+    text: str
+
+
+class DesktopNotificationIO(IOConfig):
+
+    _input: None
+    _output: Optional[DesktopNotificationOutput] = None
+
+    def __init__(self, *args: Optional[Any], **kwargs: Optional[Any]) -> None:
+        self._logger = logging.getLogger(__name__ + "DesktopNotificationIO")
+        # Not entirely sure why, but empty properties in the yaml errors
+        self._logger.info(f"DesktopNotificationIO received args - {args}")
+        self._logger.info(f"DesktopNotificationIO received kwargs - {kwargs}")
+
+    def get_inputs(self) -> Sequence[Input]:
+
+        return []
+
+    def get_outputs(self) -> Sequence[Output]:
+        if not self._output:
+            self._output = DesktopNotificationOutput()
+
+        return [self._output]
+
+
+class DesktopNotificationOutput(Output):
+
+    _engine: DesktopNotificationOutputEngine
+    _logger: logging.Logger
+
+    def __init__(self, *args: Optional[Any], **kwargs: Optional[Any]) -> None:
+        super().__init__(*args, **kwargs)
+
+        self._logger = logging.getLogger(__name__ + "DesktopNotificationOutput")
+        self._logger.info(
+            "Starting DesktopNotificationOutputEngine. Please assume crash positions!"
+        )
+
+        self._engine = DesktopNotificationOutputEngine()
+
+    @staticmethod
+    def consumes_outputs() -> Set[Type[OutputEvent]]:
+        """
+        Defines the set of output events that this Output class can consume
+        :return:
+        """
+        return {DesktopNotificationOutputEvent}
+
+    async def output(self, event: OutputEvent) -> bool:
+        """
+        Does the work of transmitting the event to the world.
+        :param event:
+        :return:
+        """
+        self._logger.info(f"output triggering with {event}")
+
+        if not isinstance(event, DesktopNotificationOutputEvent):
+            return False
+
+        return self._engine.notify(event.title, event.text)
+
+
+class DesktopNotificationOutputEngine:
+    """
+    Contains the dubious and concerning logic required to make output work on all target systems.
+    """
```

Comment:
The doc block should at least mention what this is and how it works ;)

```suggestion
    """
    Cross-Platform notification shim/engine.
    
    Works by patching the `_notify` function at init time to contain the correct implementation.
    """
```

Reply:
... I mean - that might be helpful...